config/routes.rb

```diff
@@ -1,16 +1,17 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
-  devise_for :users
+  devise_for :users,
+             controllers: {
+                 sessions: 'users/sessions',
+                 registrations: 'users/registrations'
+             }
+  
+  
 
   root "users#index"
   resources :users, only: [:index, :show, :edit, :update] do
-    resources :posts, only: [:index, :new, :show, :create, :update, :destroy]
-  end
-
-  resources :posts do
-    resources :comments, only: [:new, :create, :destroy]
-    resources :likes, only: [:create]
+    resources :posts, only: [:index, :new, :show, :create, :destroy]
   end
 
   namespace :api, defaults: { format: :json } do
```

Comment:
✅ Has API endpoint that returns all post in JSON format - **GET :** `/api/users/:user_id/posts` or `/api/v1/users/:user_id/posts` 👍🏻 

* Has API endpoint that returns all comments for a post - **GET :** `/users/:user_id/posts/:id/comments`
* Has API endpoint that adds' a comment to a post - **POST :** `/users/:user_id/posts/:id/comments`